lib/json/schema_builder/validation.rb

```diff
@@ -19,7 +19,33 @@ def fully_validate(data, opts = { })
 
       def _validate(validator, data, opts)
         opts.reverse_merge! options if options
-        JSON::Validator.send validator, as_json, data, opts
+        customize_errors = opts.delete(:customize_errors)
+        opts[:errors_as_objects] = true if customize_errors
+        validation_errors = JSON::Validator.send validator, as_json, data, opts
+        customize_errors ? _customize_errors(validation_errors) : validation_errors
+      end
+
+      def _customize_errors(error_objects)
+        _flatten_errors(error_objects).each do |error|
+          entities = Array(fragments[error[:fragment]]).select(&:error)
+          entities.each do |entity|
+            handler = entity.error
+            case handler
+            when ::Proc
+              handler.call(entities, error)
```

Comment:
I'm sure there's a reason, but why do you pass in entities to the the handler and not the single entity that the error is defined on?

Reply:
Since multiple entities can specify the same fragment, they can also specify error handlers for it and the JSON::Validator error only reports the fragment causing the error.

We basically have to pass the error to all entities that handle errors for that fragment and let the handlers determine how to deal with that.  Most of the time that just means checking the `:failed_attribute` on the error and making sure it's the expected type.

Reply:
Ah cool, makes sense